File: Level 2/택배상자/solution.py
from collections import deque


def solution(order):
    n = len(order)
    queue = deque(order)
    stack = []

    answer = 0

    for i in range(1, n + 1):
        stack.append(i)

        while stack and stack[-1] == queue[0]:
            queue.popleft()
            stack.pop()
            answer += 1

    return answer

File: Level 2/택배상자/solution.js
function solution(order) {
  const n = order.length;
  order.reverse();
  const stack = [];

  let answer = 0;

  for (let i = 1; i <= n; i++) {
    stack.push(i);

    while (
      stack.length &&
      stack[stack.length - 1] === order[order.length - 1]
    ) {
      order.pop();
      stack.pop();
      answer++;
    }
  }

  return answer;
}
